objects/src/accounts/code.rs

```diff
@@ -1,50 +1,67 @@
+use crypto::merkle::SimpleSmt;
+use miden_lib::MidenLib;
+use miden_stdlib::StdLibrary;
 
 // ACCOUNT CODE
 // ================================================================================================
 
+// CONSTANTS
+// -----------------------------------------------------------------------------------------------
+
+/// The depth of the Merkle tree that is used to commit to the account's public interface.
+const ACCOUNT_CODE_TREE_DEPTH: u8 = 8;
+
 /// Describes public interface of an account.
 ///
 /// Account's public interface consists of a set of account methods, each method being a Miden VM
 /// program. Thus, MAST root of each method commits to the underlying program. We commit to the
 /// entire account interface by building a simple Merkle tree out of all method MAST roots.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)] // TODO: Add PartialEq, Eq
 pub struct AccountCode {
-    method_tree: MerkleTree,
+    method_tree: SimpleSmt,
     module: ModuleAst,
-    // methods: Vec<Program>, commented out because Program doesn't currently implement Eq and
-    // PartialEq. Also, there might be a better way of describing a set of programs as they
-    // might share a lot of common code blocks. In a way, we want something like a Program
-    // struct but with many entry points.
+    procedures: Vec<Digest>,
 }
 
 impl AccountCode {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+    pub const ACCOUNT_CODE_PATH: &'static str = "context::account";
+
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
     /// Creates and returns a new definition of an account's interface compiled from the specified
     /// source code.
     pub fn new(source: &str) -> Result<Self, AccountError> {
-        let _module_ast = ModuleAst::parse(source)?;
+        let module_ast = ModuleAst::parse(source)?;
+        let module =
+            Module::new(LibraryPath::new(Self::ACCOUNT_CODE_PATH).expect("valid path"), module_ast);
 
-        // TODO: compile module AST into a set of program MASTs. To do this we need to expose
-        // a new method on the assembler to compile a module rather than a program (something
-        // similar to Assembler::compile_module() but without internal parameters).
-        //
-        // Open question: how to initialize the assembler? Specifically, which libraries to
-        // initialize it with. stdlib and midenlib are the two libraries we need for sure - but
-        // how to handle accounts which rely on some user-defined libraries? i.e., should there
-        // be a way to specify an "on-chain" library somehow?
+        let assembler = Assembler::default()
+            .with_library(&MidenLib::default())
+            .expect("failed to load miden-lib")
+            .with_library(&StdLibrary::default())
+            .expect("failed to load std-lib");
```

Comment:
1) Yeah this makes sense.  However I wonder if it should be an instantiated `Assembler` we keep in memory or if we should instead allow the `Assembler` to be instantiated with some pre-compiles as suggested in [this comment](https://github.com/0xPolygonMiden/miden-vm/pull/895#discussion_r1195934332).  

2) Yeah I think for now it makes sense to pass instantiated objects to the `AccountCode` constructor.  I will update it.

Reply:
Regarding the first point: I'm actually thinking of `proc_cache` in the assembler as a way to keep some pre-compiled code in memory. Once a procedure gets into that cache, it won't need to be compiled again (and right now, we put procedures into this cache on first usage - but we can also do it preemptively).

Regarding the second point: I actually meant to update the `Account` constructor to take `AccountCode` as an argument (and not change `AccountCode` constructor for now). But if you'd like to change `AccountCode` constructor now as well, that's fine too.

Reply:
> Regarding the first point: I'm actually thinking of `proc_cache` in the assembler as a way to keep some pre-compiled code in memory. Once a procedure gets into that cache, it won't need to be compiled again (and right now, we put procedures into this cache on first usage - but we can also do it preemptively).

Could this lead to unbounded growth?  Say we have a `Assembler` on the node client - would this same `Assembler` be responsible for compiling every transaction forever?  After processing a large number of transactions the `proc_cache` could get large - maybe this ins't an issue?  I was thinking that we would want some long lived cached procedures (`minden-lib` + `std` libraries) and then the account procedures etc would only be scoped to a specific transaction compilation. 

> Regarding the second point: I actually meant to update the `Account` constructor to take `AccountCode` as an argument (and not change `AccountCode` constructor for now). But if you'd like to change `AccountCode` constructor now as well, that's fine too.

Ahh ok yes this makes sense.

Reply:
> Could this lead to unbounded growth?

Yes, this could be an issue. In the longer term, I was thinking of changing `proc_cache` to something like LRU cache where if it reaches certain size, some procedures which haven't been used for a while would be evicted.

Reply:
Yeah that makes sense!